packages/contracts/contracts/optimistic-ethereum/libraries/utils/Lib_Buffer.sol

```diff
@@ -0,0 +1,251 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >0.5.0 <0.8.0;
+
+/**
+ * @title Lib_Buffer
+ * @dev Essentially just a fancy array.
+ */
+library Lib_Buffer {
+
+    /*************
+     * Libraries *
+     *************/
+
+    using Lib_Buffer for Buffer;
+
+
+    /***********
+     * Structs *
+     ***********/
+
+    struct Buffer {
+        bytes32 context;
+        mapping (uint256 => bytes32) buf;
+    }
+
+    struct BufferContext {
+        uint40 length;
+        bytes27 extraData;
```

Comment:
Would love to see some comments here explaining the nature of `extraData`.

Reply:
Done.